### PR TITLE
dev-util/gnome-builder: Version bump to 44.1, new dev-libs/libdex

### PR DESCRIPTION
**Pull request by flint2**
44.1 released, compile OK, test OK, works OK

### CODE REVIEW

**Comment by mattst88** [comment]
libpanel's dependencies need to be updated.

**Comment by flint2** [comment]
> libpanel's dependencies need to be updated.

Done in https://github.com/gentoo/gentoo/pull/30124/commits/2cda3ab2632c2f0b9822de899611658362fa1310
Thanks!

**Comment by flint2** [comment]
gnome-builder-44.rc compiles ok with webkit-gtk-2.39.7-r600 (https://github.com/gentoo/gentoo/pull/30124/commits/c2347daca46ae9b6c87a5d1b815508de3e324b83)
- detection is ok :
WebKit ................ : true (webkitgtk-6.0)
- build is ok
- test is ok
Ok:                 21  

Works ok, used to open and compile gnome-calendar project

**Comment by mattst88** [comment]
I've got webkit-gtk 2.40.0 building locally. I expect to have it in the tree tomorrow and then I'll merge the rest of this. Thanks!

**Comment by flint2** [comment]
> to have it in the tree tomorrow and then I'll

Totally missed the release because I´m focused in Gnome https://download.gnome.org/sources/?C=M&O=D

Thanks!

**Comment by flint2** [comment]
> I've got webkit-gtk 2.40.0 building locally. I expect to have it in the tree tomorrow and then I'll merge the rest of this. Thanks!

Just submitted a PR for epiphany https://github.com/gentoo/gentoo/pull/30214
Now I´m planning to test gnome-builder

File: dev-libs/libdex/libdex-0.2.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+IUSE="gtk-doc +introspection test vala"
+REQUIRED_USE="
+	gtk-doc? ( introspection )
+	vala? ( introspection )
+"
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	>=dev-libs/glib-2.68:2
```
Comment (mattst88) [inline dev-libs/libdex/libdex-0.2.0.ebuild:23]: After glib, I see this:

````
if host_machine.system() == 'linux'
  liburing_req_version = '0.7'
  liburing_req = '>= @0@'.format(liburing_req_version)
  liburing_dep = dependency('liburing', version: liburing_req, required: false)

  if liburing_dep.found()
    config_h.set10('HAVE_LIBURING', true)
  endif
else
  liburing_dep = disabler()
endif
````

Let's just require `sys-libs/liburing` unconditionally.

Following that, I see

````if get_option('sysprof')
  libsysprof_capture_dep = dependency('sysprof-capture-4')
  config_h.set10('HAVE_SYSPROF', true)
endif
````

so I think we should have `IUSE=sysprof` and a dependency on `dev-util/sysprof-capture`.

File: dev-libs/libdex/libdex-0.2.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	dev-libs/json-glib[introspection?]
```
[the remark this answers is not in the document]
Comment (mattst88) [reply dev-libs/libdex/libdex-0.2.0.ebuild:24]: I'd suggest installing and enabling `app-portage/iwdevtools`. When a package is merged, it checks dependencies and prints things like this:

````
 * VDB: detected possibly incorrect RDEPEND (dev-libs/libdex-0.2.0)
 * dev-libs/gobject-introspection:= < 
 * dev-libs/json-glib               < 
 *                                  > sys-libs/liburing:=
````

It's not 100% accurate, but it does a pretty good job.

**Review by mattst88 — CHANGES_REQUESTED** [review]
Looks really good. Nice work!

File: dev-libs/libdex/libdex-0.2.0.ebuild (hunk 2 above)
[the remark this answers is not in the document]
Comment (flint2) [reply dev-libs/libdex/libdex-0.2.0.ebuild]: > > Work in Progress, do not merge, not compiling
> 
> Is this the error you saw?
> 
> ```
> ../gnome-builder-44.0/src/libide/gui/ide-application-tweaks.c: In function ‘create_plugin_toggle’:
> ../gnome-builder-44.0/src/libide/gui/ide-application-tweaks.c:65:3: error: implicit declaration of function ‘adw_expander_row_add_suffix’; did you mean ‘adw_expander_row_add_prefix’? [-Werror=implicit-function-declaration]
>    65 |   adw_expander_row_add_suffix (row, GTK_WIDGET (toggle));
>       |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~
>       |   adw_expander_row_add_prefix
> ../gnome-builder-44.0/src/libide/gui/ide-application-tweaks.c:65:3: warning: nested extern declaration of ‘adw_expander_row_add_suffix’ [-Wnested-externs]
> cc1: some warnings being treated as errors
> ```
> 
> If so, looks like it should be fixed by https://gitlab.gnome.org/GNOME/gnome-builder/-/commit/d89a5c95e43f4a4e1a4e9d063e3962bac8e51db0

Now with 44.1 is compiling OK, Ready for review

**Pull request closed without merging**